src/utils/hostname.ts

```diff
@@ -0,0 +1,31 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+// return Top Level Domain from URL
+export function extractHostname(url: string, tld: boolean) {
```

Comment:
I would strongly recommend using the NodeJS URL `hostname` API instead.
https://nodejs.org/api/url.html#url_url_hostname